### PR TITLE
Better enforce that the rest response contains valid JSON

### PR DESCRIPTION
**Pull request by lava**

### CODE REVIEW

File: libvast/src/http_api.cpp [hunk 1]
```diff
@@ -8,12 +8,34 @@
...
 #include <simdjson.h>
 
+namespace {
+
+// Validates that the argument is a valid JSON object.
+auto validate_json(const std::string& json) -> bool {
+  auto parser = simdjson::dom::parser{};
+  auto padded = simdjson::padded_string{json};
+  auto obj = simdjson::dom::object{};
+  return parser.parse(json).get(obj) == simdjson::SUCCESS;
```
Comment (jachris) [inline libvast/src/http_api.cpp:25]: This does not work, because `simdjson` does lazy validation.

Comment (lava) [reply libvast/src/http_api.cpp:25]: Also for the `dom` parser? I thought it was only for the `ondemand` parser.

Comment (jachris) [reply libvast/src/http_api.cpp:25]: Sorry, you are correct :+1: I thought it would be `ondemand`, but you are using `dom`.

**Review by jachris — APPROVED** [review]

File: libvast/src/http_api.cpp [hunk 2]
```diff
@@ -223,7 +254,8 @@ auto rest_response::make_error(uint16_t error_code, std::string message,
 
 auto rest_response::make_error_raw(uint16_t error_code, std::string body,
                                    caf::error detail) -> rest_response {
-  auto result = rest_response{std::move(body)};
+  auto result = rest_response{};
+  result.body_ = std::move(body);
```
Comment (jachris) [inline libvast/src/http_api.cpp:258]: I think this would also benefit from a validity check.

**Pull request merged**